### PR TITLE
Override entrypoint cmd

### PR DESCRIPTION
**Pull request by waakanni**
I had to alter the deployment file as well

`helm template access-request`
this is the rendered output:
``` yaml 
spec:
      serviceAccountName: release-name
      shareProcessNamespace: false
      containers:
      - image: "ghga/access-request-service:2.0.3"
        command: ["sh", "-c"]
        args:
          - if [ -d "/vault/secrets" ]; then
              for f in /vault/secrets/*; do
                [ -f "$f" ] && . "$f";
              done
            fi;
            - "run-rest"
        name: release-name
        securityContext:

### CODE REVIEW

**Comment by ckaipf** [comment]
Thanks @waakanni for picking that up so quickly! I see that is tricky since we need to source the Vault secrets before running the command. Did you try to just use a string there, not an array? So 
```yaml
image:
 command: "cmd --arg1 --arg2"
```

And in template

```yaml
command: ["sh", "-c"]
args:
  - if [ -d "/vault/secrets" ]; then
      for f in /vault/secrets/*; do
        [ -f "$f" ] && . "$f";
      done
    fi;
  - {{ .Values.image.command }}
```
I guess that should be okay like this and is maybe more readable.

**Comment by waakanni** [comment]
> Thanks @waakanni for picking that up so quickly! I see that is tricky since we need to source the Vault secrets before running the command. Did you try to just use a string there, not an array? So
> 
> ```yaml
> image:
>  command: "cmd --arg1 --arg2"
> ```
> 
> And in template
> 
> ```yaml
> command: ["sh", "-c"]
> args:
>   - if [ -d "/vault/secrets" ]; then
>       for f in /vault/secrets/*; do
>         [ -f "$f" ] && . "$f";
>       done
>     fi;
>   - {{ .Values.image.command }}
> ```
> 
> I guess that should be okay like this and is maybe more readable.

I tried this but it throws an error.
`Error: template: access-request/templates/deployment.yml:35:12: executing "access-request/templates/deployment.yml" at <include "ghga-common.command-args" (list "ars")>: error calling include: template: access-request/charts/ghga-common/templates/_command-args.tpl:15:14: executing "ghga-common.command-args" at <.Values.image.command>: can't evaluate field Values in type []interface {}
helm.go:84: [debug] template: access-request/templates/deployment.yml:35:12: executing "access-request/templates/deployment.yml" at <include "ghga-common.command-args" (list "ars")>: error calling include: template: access-request/charts/ghga-common/templates/_command-args.tpl:15:14: executing "ghga-common.command-args" at <.Values.image.command>: can't evaluate field Values in type []interface {}`

**Comment by ckaipf** [comment]
Could you try
`ghga-common.command-args . `
instead of 
`ghga-common.command-args" (list "ars")`?
I guess you need to pass the dot environment to the include template function.

**Comment by waakanni** [comment]
> Could you try `ghga-common.command-args . ` instead of `ghga-common.command-args" (list "ars")`? I guess you need to pass the dot environment to the include template function.

Ah I thought you wanted keep the `(list "ars")`. That solves the problem then

**Comment by waakanni** [comment]
@ckaipf do you want me to add this to all of our charts?

File: charts/access-request/values.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ horizontalPodAutoscaler:
   averageMemoryUtilization: 75
 image:
   name: "ghga/access-request-service"
+  command: "cmd --arg1 --arg2"
```
Comment (ckaipf) [inline charts/access-request/values.yaml:9]: Please check this.

**Comment by ckaipf** [comment]
> @ckaipf do you want me to add this to all of our charts?

Yes, this needs to be updated in every chart. Please provide a end-to-end test before, we are currently in a pre-launch phase and should therefore avoid merging untested changes.

**Comment by waakanni** [comment]
@ckaipf some charts are calling the cmd template multiple time. this was my solution
this is the json output of the deployed 'work-package' service

```
 {
                        "args": [
                            "if [ -d \"/vault/secrets\" ]; then for f in /vault/secrets/*; do [ -f \"$f\" ] \u0026\u0026 . \"$f\"; done fi;",
                            "run-rest, consume-events"
                        ],
                        "command": [
                            "sh",
                            "-c"
                        ],

**Comment by ckaipf** [comment]
You are right, there are some pods which run two containers. You'll need to figure out a way how to implement this. The proposed solution is not correct. You could try defining different values, or hard code the command e.g. for the rest container.

**Pull request closed without merging**